contributors/design-proposals/node/troubleshoot-running-pods.md

```diff
@@ -356,38 +393,47 @@ cluster administrators to easily constraint debug container images.
     but Debug Containers would benefit from future improvements or work arounds.
-1.  We do not want to describe Debug Containers using `v1.Container`. This is to
-    reinforce that Debug Containers are not general purpose containers by
-    limiting their configurability. Debug Containers should not be used to build
-    services.
-1.  Debug Containers are of limited usefulness without a shared PID namespace.
-    If a pod is configured with isolated PID namespaces, the Debug Container
+1.  Debug Containers should not be used to build services, which we've attempted
+    to reflect in the API.
+1.  If a pod is configured with isolated PID namespaces, the Debug Container
     will join the PID namespace of the target container. Debug Containers will
-    not be available with runtimes that do not implement PID namespace sharing
-    in some form.
+    not be available with runtimes that do not implement PID namespace sharing.
 
 ## Implementation Plan
 
 ### Alpha Release
 
 #### Goals and Non-Goals for Alpha Release
 
-We're targeting an alpha release in Kubernetes 1.9 that includes the following
+We're targeting an alpha release in Kubernetes 1.10 that includes the following
 basic functionality:
 
 *   Support in the kubelet for creating debug containers in a running pod
-*   A `kubectl debug` command to initiate a debug container
+*   A `kubectl alpha debug` command to initiate a debug container
 *   `kubectl describe pod` will list status of debug containers running in a pod
 
 Functionality will be hidden behind an alpha feature flag and disabled by
-default. The following are explicitly out of scope for the 1.9 alpha release:
+default. The following are explicitly out of scope for the alpha release, but
+must be resolved prior to beta release:
+
+*   There's no guarantee that exited Debug Containers won't be garbage collected
+    as regular containers, so they may disappear from the list of
+    `EphemeralContainers`.
+*   We could improve reliability of `UpdatePodOptions.OnCompleteFunc` by
+    prioritizing based on `SyncPodType`.
 
-*   Exited Debug Containers will be garbage collected as regular containers and
-    may disappear from the list of Debug Container Statuses.
-*   Security Context for the Debug Container is not configurable. It will always
-    be run with `CAP_SYS_PTRACE` and `CAP_SYS_ADMIN`.
-*   Image pull policy for the Debug Container is not configurable. It will
-    always be run with `PullAlways`.
+#### Kubernetes API Changes
+
+The following changes must be implemented in the API:
+
+1.  `v1.EphemeralContainer` will be added and `v1.PodStatus` will be extended as
+    described above.
+1.  The new subresource will be added to the pods api, validation added and
+    proxied to the kubelet.
+1.  The API server must check for Ephemeral Containers when validating `attach`.
+
+New fields will be prepended with `alpha` for the duration of this feature's
```

Comment:
remove this... field name prefixing is not recommended in https://github.com/kubernetes/community/blob/master/contributors/devel/api_changes.md#adding-unstable-features-to-stable-versions

Reply:
Fixed.